test/framework/clusterstatevalidations/cluster.go

```diff
@@ -0,0 +1,186 @@
+						workerGroupCount++
+						if err := validateNodeReady(node, vc.ClusterSpec.Cluster.Spec.KubernetesVersion); err != nil {
+							return fmt.Errorf("failed to validate worker node ready %v", err)
+						}
+					}
+				}
+			}
+		}
+		if workerGroupCount != *w.Count {
+			return fmt.Errorf("worker node group %s count does not match expected: %d of %d", w.Name, workerGroupCount, *w.Count)
+		}
+	}
+	return nil
+}
+
+// ValidateClusterDoesNotExist checks that the cluster does not exist by attempting to retrieve the CAPI cluster.
+func ValidateClusterDoesNotExist(ctx context.Context, vc clusterf.StateConfig) error {
+	clus := vc.ClusterSpec.Cluster
+	capiCluster, err := controller.GetCAPICluster(ctx, vc.ManagementClusterClient, clus)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve cluster %s", err)
+	}
+	if capiCluster != nil {
+		return fmt.Errorf("cluster %s exists", capiCluster.Name)
+	}
+	return nil
+}
+
+// ValidateCilium gets the cilium-config from the cluster and checks that the cilium
+// policy in cluster.Spec matches the enabled policy in the config.
+func ValidateCilium(ctx context.Context, vc clusterf.StateConfig) error {
+	clusterClient := vc.ClusterClient
+
+	yaml := vc.ClusterSpec.Cluster
+	cm := &corev1.ConfigMap{}
+	key := types.NamespacedName{Namespace: "kube-system", Name: "cilium-config"}
+	err := clusterClient.Get(ctx, key, cm)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve configmap %s", err)
+	}
+
+	clusterCilium := cm.Data["enable-policy"]
+	yamlCilium := string(yaml.Spec.ClusterNetwork.CNIConfig.Cilium.PolicyEnforcementMode)
+	if yamlCilium == "" && clusterCilium == "default" {
+		return nil
+	}
+	if clusterCilium != yamlCilium {
+		return fmt.Errorf("cilium policy does not match. ConfigMap: %s, YAML: %s", clusterCilium, yamlCilium)
+	}
+
+	return nil
+}
+
+func validateNodeReady(node corev1.Node, kubeVersion v1alpha1.KubernetesVersion) error {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == "Ready" && condition.Status != corev1.ConditionTrue {
+			return fmt.Errorf("node %s not ready yet. %s", node.GetName(), condition.Reason)
+		}
+	}
+	kubeletVersion := node.Status.NodeInfo.KubeletVersion
+	if !strings.Contains(kubeletVersion, string(kubeVersion)) {
+		return fmt.Errorf("validating node version: kubernetes version %s does not match expected version %s", kubeletVersion, kubeVersion)
+	}
+	return nil
+}
+
+// getWorkerNodeMachineSets gets a list of MachineSets corresponding the provided WorkerNodeGroupConfiguration from the management cluster.
+func getWorkerNodeMachineSets(ctx context.Context, vc clusterf.StateConfig, w v1alpha1.WorkerNodeGroupConfiguration) ([]v1beta1.MachineSet, error) {
+	md := &v1beta1.MachineDeployment{}
+	mdName := fmt.Sprintf("%s-%s", vc.ClusterSpec.Cluster.Name, w.Name)
+	key := types.NamespacedName{Name: mdName, Namespace: constants.EksaSystemNamespace}
+	if err := vc.ManagementClusterClient.Get(ctx, key, md); err != nil {
```

Comment:
why do we need to read the machine deployment if the only thing we need is its name? We are using the name to read it, so we are not gaining anything from that extra call

Reply:
i also meant it as a check to make sure that the corresponding machine deployment exists. do you think that's unnecessary for this validation?

Reply:
IMO that is an implementation detail. This validations are checking that the state of the cluster complies with the cluster specification. Machine deployments are not part of the cluster state, they are a mean to manage/control that state.